user-guide/Advanced_Functionality/Databases/Supported_System_Data_Storage_Architectures/Resource_migration_to_elastic.md

```diff
@@ -0,0 +1,139 @@
+
+When using ``Xml`` as storage, it is not allowed to remove a resource while one of the agents in the cluster is unreachable, as this could cause syncing issues. When using Elasticsearch as storage, this is allowed.
+
+### Initial event on subscriptions
+
+When using ``Xml`` as storage, if a client subscribes on the ``ResourceManagerEventMessage``, he will receive an initial event with all resources and resource pools. When using Elasticsearch as storage these events will no longer be sent. This is to prevent sending a message to the client containing thousands of resources.
+
+### Syncing
+
+When using ``Xml`` storage, all ResourceManagers in the cluster will sync the resources in their ``Xml`` file on startup and during the midnight sync. When using Elasticsearch as storage, DataMiner relies on Elasticsearch to do the syncing and data will not be sent around during the midnight sync or during startup. The ResourceManager will refresh the cached resources during the midnight sync by reading all resources and resource pools again from Elasticsearch.
+
+## Metrics
+
+### Resource performance
+
+Below you can find the result of our performance testing for resource add, update and delete operations. On our setup (with a local Elasticsearch), Elasticsearch started to outperform ``Xml`` storage when more than 2000 resources were present on the system.
+
+![SRM Resource create performance](~/user-guide/images/SRM_Resource_Create_Performance.png)
+![SRM Resource update performance](~/user-guide/images/SRM_Resource_Update_Performance.png)
+![SRM Resource delete performance](~/user-guide/images/SRM_Resource_Delete_Performance.png)
+
+__Note__: read performance of resources and resource pools is unchanged, since all reads will go to a cache in SLNet. This was also confirmed during our testing.
+
+### ResourceManager startup performance
+
+See the following table for performance numbers for the startup time for ResourceManager with a large amount of resources:
+
+| __Storage type__ 	| __Resource cache load time__ 	| __ResourceManager startup time__ 	|
+|------------------	|------------------------------	|----------------------------------	|
+| Elasticsearch    	| 34s                          	| 1m 45s                           	|
+| Xml              	| 47s                          	| 1m 52s                           	|
+
+The __Resource cache load time__ columns shows the time it takes for ResourceManager to load all resources in its cache on startup. The __ResourceManager startup time__ is the total time until ResourceManager is operational after initial startup. This includes the time it takes to initialize all storage types and can be higher if a lot of bookings are currently active or will start soon. This does not include the time ResourceManager spends syncing with other agents in the cluster if ``Xml`` storage is used. The ResourceManager is considered operational before it has fully synced, since it can start handling API requests.
```

Comment:
```suggestion
The __Resource cache load time__ column shows the time it takes for ResourceManager to load all resources in its cache on startup. The __ResourceManager startup time__ is the total time until ResourceManager is operational after initial startup. This includes the time it takes to initialize all storage types and can be higher if a lot of bookings are currently active or will start soon. This does not include the time ResourceManager spends syncing with other agents in the cluster if ``Xml`` storage is used. The ResourceManager is considered operational before it has fully synced, since it can start handling API requests.
```